### PR TITLE
Modify Readme to include Authentication Details

### PR DESCRIPTION
**Pull request by sarangan12**
This PR is to address Readme changes listed in Issue https://github.com/Azure/autorest.typescript/issues/721

@ramya-rao-a Please review and approve

### CODE REVIEW

**Review by deyaaeldeen — APPROVED** [review]
Looks good!

**Review by joheredi — APPROVED** [review]

File: src/generators/static/readmeFileGenerator.ts [hunk 1]
```diff
@@ -27,10 +27,12 @@ npm install ${packageDetails.name}
 \`\`\`
 
 ### How to use
+#### Authentication
+One of the first steps in accessing the services using SDK is to authenticate the client. There are several methods to authenticate such as interactive login, etc. Refer [readme for @azure/ms-rest-nodeauth](https://www.npmjs.com/package/@azure/ms-rest-nodeauth) package for all options to authenticate the client.
```
Comment (ramya-rao-a) [inline src/generators/static/readmeFileGenerator.ts:31]: This PR is against the v6 branch.

From my understanding the packages generated using v6 of the code generator use `TokenCredential` and therefore a user will have to bring in `@azure/identity`. Here, we are talking about `@azure/ms-rest-nodeauth` which is for the packages generated using v4 of the code generator.

Comment (ramya-rao-a) [reply src/generators/static/readmeFileGenerator.ts:31]: The issue https://github.com/Azure/autorest.typescript/issues/721 is mainly for packages generated using the v4 code generator. The packages generated using the v6 generator are getting the convenience layer and hand written readmes anyway and so does not have this immediate need.

Comment (sarangan12) [reply src/generators/static/readmeFileGenerator.ts:31]: @ramya-rao-a  Does the below line makes sense? 

```
One of the first steps in accessing the services using SDK is to authenticate the client. There are several methods to authenticate such as DefaultAzureCredential, etc. Refer [readme for @azure/identity](https://www.npmjs.com/package/@azure/identity) package for all options to authenticate the client.
```

**Comment by sarangan12** [comment]
Based on the above conversation, I dont think this change is required anymore. Closing this for now

**Comment by deyaaeldeen** [comment]
@sarangan12 I think we still should at least fix the link to the samples.

**Pull request closed without merging**